kpm/lib/kpm/installer.rb

```diff
@@ -99,5 +100,43 @@ def install_ruby_plugins(force_download, verify_sha1)
 
       infos
     end
+
+    def clean_up_descriptors
+      removed_plugins = clean_up_plugin_identifiers
+      clean_up_sha1s(removed_plugins)
+    end
+
+    def clean_up_plugin_identifiers
+      inspector = KPM::Inspector.new
+      installed_plugins = inspector.inspect(@config['plugins_dir'])
+
+      plugins_installation_path = File.join(@config['plugins_dir'], 'plugins')
+      plugins_manager = KPM::PluginsManager.new(plugins_installation_path, @logger)
+
+      plugin_identifiers = plugins_manager.read_plugin_identifiers
+      removed_identifiers = []
```

Comment:
That works (A maybe slightly better approach would have been to modify the inspector to already know which plugins appear in the `plugin_identifiers.json` to avoid having to rely on the `KPM::PluginsManager` to pull the entries).